tests/unit_tests/test_routes.py

```diff
@@ -153,9 +185,35 @@ def test_update_non_existing_kml(self):
     def test_valid_kml_put_non_allowed_origin(self):
         id_to_put = self.sample_kml['id']
         response = self.app.put(
-            f'/kml/{id_to_put}',
-            data=self.kml_dict["updated"],
-            content_type="application/vnd.google-earth.kml+xml",
+            url_for('update_kml', kml_id=id_to_put),
+            data=prepare_kml_payload(
+                self.kml_dict["updated"], admin_id=self.sample_kml['admin_id']
+            ),
+            content_type="multipart/form-data",
+            headers=self.origin_headers["bad"]
+        )
+        self.assertEqual(response.status_code, 403)
+        self.assertEqual(response.content_type, "application/json")
+        self.assertEqual(response.json["error"]["message"], "Not allowed")
+
+    def test_valid_kml_put_non_allowed_admin_id(self):
+        id_to_put = self.sample_kml['id']
+        response = self.app.put(
+            url_for('update_kml', kml_id=id_to_put),
+            data=prepare_kml_payload(self.kml_dict["updated"], admin_id='invalid-id'),
+            content_type="multipart/form-data",
+            headers=self.origin_headers["bad"]
+        )
+        self.assertEqual(response.status_code, 403)
+        self.assertEqual(response.content_type, "application/json")
+        self.assertEqual(response.json["error"]["message"], "Not allowed")
```

Comment:
Picky, I know: Later on you use "Permission denied" for non allowed admin_id operations. (just for consistency)